### DISCUSSION TITLE
[Feature Request] Some way to allow all mods in kbx

### DISCUSSION THREAD

**Discussion by Lighthez** [TeamRizu/OutFox#552] [03 - Feature Requests]
Currently, most mods are disabled in the kbx gamemode. I would like to write modfiles for this specific mode, but this restriction prohibits me from adding attacks or triggering mods via lua. 

Using another gamemode would result in confusion from players, who would most likely have a mixture of "normal" files that are playable with the native control scheme of the mode, and files that are possibly unplayable in the native control scheme. Kbx with modfiles would allow creators to have a standardized way of writing "keyboard modfiles" that behave consistently with any amount of keys. 

As was pointed out, kbx was designed with keyboard players (players of osu!mania, Etterna and other keyboard vsrgs) in mind and mods aren't a typical feature of these games. The mode has better performance thanks to the absence of mods, and this covers a large portion of kbx usage.

With these problems in mind, an ideal solution would retain the performance benefit of having mods disabled when they are not needed, and the creative freedom and convenience of having mods for a keyboard-based mode. One solution (although very hacky) would be to modify the .sm and .scc formats to specify a field that enables or disabled mods for the charts that need them. This solution isn't ideal, but allows other modes to gain the same performance benefits that having mods disabled entails.

Any other solutions would be appreciated, as there are probably less complicated ways of achieving this feature.

**Comment by Sudospective** [comment]
As far as I know, this was something that was not only done intentionally, but was specifically noted as something that would stay. I think if mods were to be enabled for KBX, a rewrite for the mod system would be an ideal prerequisite. A lot of people playing stuff like Etterna and osu!mania seem like they would prefer performance over features, and I think that was the intention with the mode. That being said, the only column count I don't think has a viable way to be modded is 1, 2, and 7 columns and some counts higher than 9, but all others can be modded (3k, 4k, and 6k from dance, 5k from pump, 8k and 9k from techno). Even then, these column counts can be done if you know your way around Lua and the game mode has enough columns. It would just be ideal to specifically mark the chart as a keyboard chart rather than a pad chart if you plan on using those modes as a workaround.

I wonder how `Def.NoteField` behaves in KBX 🤔
